ch05.adoc

```diff
@@ -747,3 +747,291 @@ data:
 
 In this example both the analysis time and the single pressure level are represented using scalar coordinate variables. The analysis time is identified by the standard name "forecast_reference_time" while the valid time of the forecast is identified by the standard name "time".
 
+
+[[domain-variables, Section 5.8, "Domain Variables"]]
+=== Domain Variables
+
+A domain describes data locations and cell properties. It defines
+cells that span a collection of dimensions with cell coordinates, cell
+measures, and coordinate reference systems.
+
+A data variable defines its domain via its own attributes, but a
+domain variable provides the description of a domain in the absence of
+any data values. It is of arbitrary type since it contains no data. It
+acts as a container for the attributes that define the domain. The
+purpose of a domain variable is to provide domain information to
+applications that have no need of data values at the domain's
+locations, thus removing any ambiguity when retrieving a domain from a
+dataset. Ancillary variables and cell methods are not part of the
+domain, because they are only defined in relation to data values.
+
+The domain variable supports the same attributes as are allowed on a
+data variable for describing a domain, with exactly the same meanings
+and syntaxes, as described in <<attribute-appendix>>. If an attribute
+is needed by a particular data variable to describe its domain, then
+that attribute would also be needed by the equivalent domain variable.
```

Comment:
Having domain variables adopt from field variables in this way may become problematic. I think this is probably the right -- pragmatic -- approach for now, but we need to be careful with sweeping statements like is made here.

Reply:
I appreciate your concern, here. I feel that the approach is guaranteed to work for as long as CF still insists that a data variable encodes is own, implicit domain, which is to say for the foreseeable future. 

My one point of concern, however, is for a scalar data variable that innocently has the `dimensions` attribute - how can we tell if it is a data variable or a domain variable?

Perhaps using the `dimensions` attribute as the identifying feature could be replaced by the existing `cf_role` attribute. I think that this could be a bit safer:
``` 
 char domain ;
        domain:cf_role = "domain" ;
        domain:dimensions = "x y" ;
        domain:coordinates = "lat lon" ;

Reply:
Resolved in https://github.com/cf-convention/cf-conventions/issues/301#issuecomment-702695406